richtextfx/src/main/java/org/fxmisc/richtext/EditableStyledDocument.java

```diff
@@ -19,7 +19,12 @@
 
     ReadOnlyStyledDocument<PS, S> snapshot();
 
-    EventStream<PlainTextChange> plainChanges();
+    default EventStream<PlainTextChange> plainChanges() {
+        return richChanges()
+                // only allow changes that added/removed text; exclude style changes
+                .filter(c -> c.insertedLength() != 0 || c.removedLength() != 0)
+                .map(c -> new PlainTextChange(c.position, c.removed.getText(), c.inserted.getText()));
+    }
```

Comment:
Yes, slightly. On the other hand there is a slight repetition of code in calling `getText()`.

And if you want to avoid an intermediate `EventStream`, use `filterMap`.